src/ui/pages/WalletSelect/WalletList.tsx

```diff
@@ -76,121 +86,154 @@ function WalletListItem({
+                                  event.stopPropagation();
+                                  copyButtonRef.current.click();
+                                }
+                              }}
+                            >
+                              {` · ${truncateAddress(wallet.address, 5)}`}
+                            </span>
+                          </>
+                        ) : null}{' '}
+                        {useCssAnchors ? (
+                          <span
+                            // This is a "slot" where copyButton will visually appear
+                            style={{
+                              display: 'inline-block',
+                              width: COPY_BUTTON_SIZE,
+                              height: COPY_BUTTON_SIZE,
+                              ['anchorName' as string]: anchorName,
+                              verticalAlign: 'bottom',
+                            }}
+                          ></span>
+                        ) : (
+                          copyButton
+                        )}
+                      </>
+                    )}
+                  />
+                </UIText>
+              }
+              detailText={
+                <PortfolioValue
+                  address={wallet.address}
+                  render={(query) => (
+                    <UIText kind="headline/h3">
+                      {query.data ? (
+                        <NeutralDecimals
+                          parts={formatCurrencyToParts(
+                            query.data.data?.totalValue || 0,
+                            'en',
+                            currency
+                          )}
+                        />
                       ) : (
-                        copyButton
+                        NBSP
                       )}
-                    </>
+                    </UIText>
                   )}
                 />
-              </UIText>
-            }
-            detailText={
-              <PortfolioValue
-                address={wallet.address}
-                render={(query) => (
-                  <UIText kind="headline/h3">
-                    {query.data ? (
-                      <NeutralDecimals
-                        parts={formatCurrencyToParts(
-                          query.data.data?.totalValue || 0,
-                          'en',
-                          currency
-                        )}
-                      />
-                    ) : (
-                      NBSP
-                    )}
-                  </UIText>
-                )}
-              />
-            }
-          />
-          {isSelected ? <CheckIcon style={{ width: 24, height: 24 }} /> : null}
-        </HStack>
+              }
+            />
+            {isSelected ? (
+              <CheckIcon style={{ width: 24, height: 24 }} />
+            ) : null}
+          </HStack>
+          {exploreRewardsUrl ? (
```

Comment:
It looks like it is better not to extend this component's signature even more, but trying to pass this layout with some prop like `renderItemFooter`. But this is debatable

Reply:
agree, your suggestion looks more general